xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/plugin/rightsmanager/RightsManagerTest.java

```diff
@@ -0,0 +1,162 @@
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.test.annotation.AllComponents;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.test.MockitoOldcore;
+import com.xpn.xwiki.test.junit5.mockito.InjectMockitoOldcore;
+import com.xpn.xwiki.test.junit5.mockito.OldcoreTest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doReturn;
+
+/**
+ * Test for the RightsManager
+ *
+ * @since 13.8
+ * @version $Id$
+ */
+@OldcoreTest
+@AllComponents
+public class RightsManagerTest
+{
+    @InjectMockitoOldcore
+    MockitoOldcore oldcore;
+
+    /**
+     * instance under test
+     */
+    private RightsManager rights;
+
+    @BeforeEach
+    protected void beforeEach() throws Exception
+    {
+        rights = RightsManager.getInstance();
+
+        doReturn(true).when(oldcore.getSpyXWiki()).checkAccess(any(), any(), any());
+    }
+
+    @Test
+    public void testRenameGroup() throws Exception
+    {
+        DocumentReference oldGroupName = new DocumentReference("xwiki", Collections.singletonList("XWiki"),
+            "GroupBefore");
+        DocumentReference newGroupName = new DocumentReference("xwiki", Collections.singletonList("XWiki"),
+            "GroupAfter");
+        String oldValue = "XWiki.GroupBefore,XWiki.Some\\,OtherGroup";
+        String expectedValue = "XWiki.Some\\,OtherGroup,XWiki.GroupAfter";
+        // ThreadLocal: only to have this final and use it in the answer
+        final ThreadLocal<String> actualValue = new ThreadLocal<String>();
+
+        XWikiDocument rightsDocument = createMockDocument();
+
+        doReturn(Collections.singletonList(rightsDocument)).when(oldcore.getMockStore()).searchDocuments(any(), any(),
+            eq(oldcore.getXWikiContext()));
+
+        BaseObject rightsObject = Mockito.mock(BaseObject.class);
+        doReturn(Collections.singletonList(rightsObject)).when(rightsDocument)
+            .getXObjects(MockitoOldcore.GLOBAL_RIGHTS_CLASS);
+        doReturn(oldValue).when(rightsObject).getLargeStringValue("groups");
+        doAnswer(new Answer<Void>()
+        {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable
+            {
+                actualValue.set((String) invocation.getArgument(1));
+                return null;
+            }
+        }).when(rightsObject).setLargeStringValue(eq("groups"), any());
+
+        rights.replaceUserOrGroupFromAllRights(oldGroupName, newGroupName, false, oldcore.getXWikiContext());
+
+        assertEquals(expectedValue, actualValue.get());
```

Comment:
This looks misleading to me since actualValue is something computed by the test and not the code under tests.

AFAICS this test only verifies that `setLargeStringValue` is called with first parameter being `groups` and a second parameter (any value). If this is what you want to do, you should probably use a `verify` instead.

Reply:
hmm my bad the value passed to `setLargeStringValue` is actually captured and set in another variable (`actualValue)` so it looks ok.

Reply:
To be clear the test wants to verify the value passed to the call to `setLargeStringValue`.

Reply:
BTW since the test exercises `replaceUserOrGroupFromAllRights`, it should probably be named `replaceUserOrGroupFromAllRights` instead of `testRenameGroup`, following our best practice at https://dev.xwiki.org/xwiki/bin/view/Community/Testing/JavaUnitTesting/#HBestpractices